engine/runtime/src/main/java/org/enso/interpreter/runtime/ThreadExecutors.java

```diff
@@ -30,6 +33,22 @@ final ExecutorService newFixedThreadPool(int cnt, String name, boolean systemThr
   }
 
   public void shutdown() {
+    synchronized (pools) {
+      shutdownPools();
+    }
+    synchronized (threads) {
+      for (var t : threads.keySet()) {
```

Comment:
Looks like this fixes the problem where `context.close()` complains about dangling threads. Probably `shutdownPool()` stops the pools, but the threads are _left to finish later_. Sometimes the `context.close()` is called faster and some of the threads are **not yet finished**. This `t.join()` seems to fix that situation.